balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -622,3 +682,57 @@ func equalAddressIgnoringBalAttributes(a, b *resolver.Address) bool {
 		a.Attributes.Equal(b.Attributes) &&
 		a.Metadata == b.Metadata
 }
+
+// callbackScheduleris used to schedule the execution of a callback after a
+// a specified delay. It is not safe for concurrent access.
+type callbackScheduler struct {
+	cancelScheduled func()
+	closed          bool
+	wg              sync.WaitGroup
+}
+
+// schedule schedules the execution of a callback. It cancels any previously
```

Comment:
> It cancels any previously scheduled callbacks

We can only have one pending scheduled callback, right?

What is the real purpose/advantage of having this type over simply using `time.AfterFunc` and storing the returned timer in the LB policy and calling `Stop` on it when you want to cancel pending callback?